src/libraries/System.Net.Security/src/System/Net/Security/SslStream.Implementation.cs

```diff
@@ -1083,7 +1078,7 @@ private ValueTask<int> FillHandshakeBufferAsync<TIOAdapter>(TIOAdapter adapter,
                 int bytesRead = t.Result;
                 if (bytesRead == 0)
                 {
-                    return new ValueTask<int>(0);
+                    throw new IOException(SR.net_io_eof);
```

Comment:
The general pattern we use for these sorts of "Fill" routines is:

(1) Return 0 if EOF is hit and no data is in the buffer -- i.e. no partial frame is received. This allows the caller to decide if receiving an immediate EOF is an error or not -- it may instead simply indicate graceful shutdown -- and if it is an error, to generate the appropriate exception, since it may be desirable to generate a different exception or message in the EOF case vs the partial frame case.
(2) Throw something like "unexpected EOF" when EOF is hit and there is data in the buffer. This means we received a partial frame, and is always an error.

I believe this is what the calling code expects -- note it checks for 0 bytes on the first call to this, but not on the second, since at that point we must have at least some data in the buffer.

Reply:
I don't think the presence of data is really relevant here. We call it possibly twice: once with just header size and then with promised frame size. In all cases, we really want to throw IOException if we cannot get enough data to continue. 

We already do that if the read does not finish synchronously here:
https://github.com/dotnet/runtime/blob/d0631be3a1b9e5a8adf16f4885778179a8ed365b/src/libraries/System.Net.Security/src/System/Net/Security/SslStream.Implementation.cs#L1101

Reply:
Okay, fair enough,

With this change, it looks like we aren't using the return value at all. Can we change it from `ValueTask<int>` to `ValueTask`?

Reply:
updated. I also removed the dead function since that is super easy.